File: GridView/CustomCellEditor/CustomCellEditorVB/CustomCellEditor/CustomCell.vb
Imports Telerik.WinControls.UI

Public Class CustomInfoCell
    Inherits GridComboBoxCellElement

    Private buttonWidth As Integer = 20
    Private buttonPadding As Integer = 2

    Public Sub New(ByVal column As GridViewColumn, ByVal row As GridRowElement)
        MyBase.New(column, row)
    End Sub

    Protected Overrides ReadOnly Property ThemeEffectiveType() As System.Type
        Get
            Return GetType(GridComboBoxCellElement)
        End Get
    End Property

    Protected Overrides Function ArrangeOverride(ByVal finalSize As SizeF) As SizeF
        MyBase.ArrangeOverride(finalSize)
        Dim rect As RectangleF = GetClientRectangle(finalSize)
        Dim rectEdit As RectangleF = New RectangleF(rect.X, rect.Y, rect.Width - (buttonWidth + buttonPadding), rect.Height)
        Dim rectButton As RectangleF = New RectangleF(rectEdit.Right + buttonPadding, rectEdit.Y, buttonWidth, rect.Height)
        If Me.Children.Count = 2 Then
            Me.Children(0).Arrange(rectButton)
            Me.Children(1).Arrange(rectEdit)
        End If

        Return finalSize
    End Function
End Class

File: GridView/CustomCellEditor/CustomCellEditorCS/CustomCellEditor/CustomCell.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;
using System.Drawing;

namespace CustomCellEditor
{
    public class CustomInfoCell : GridComboBoxCellElement
    {
        private int buttonWidth = 20;
        private int buttonPadding = 2;

        public CustomInfoCell(GridViewColumn column, GridRowElement row)
            : base(column, row)
        {            
        }

        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(GridComboBoxCellElement);
            }
        }

        protected override SizeF ArrangeOverride(SizeF finalSize)
        {
            base.ArrangeOverride(finalSize);
            RectangleF rect = GetClientRectangle(finalSize);
            RectangleF rectEdit = new RectangleF(rect.X, rect.Y, rect.Width - (buttonWidth + buttonPadding), rect.Height);
            RectangleF rectButton = new RectangleF(rectEdit.Right + buttonPadding, rectEdit.Y, buttonWidth, rect.Height);
            if (this.Children.Count == 2)
            {
                this.Children[0].Arrange(rectButton);
                this.Children[1].Arrange(rectEdit);
            }

            return finalSize;
        }
    }
}